build.md

```diff
@@ -0,0 +1,145 @@
+```bash
+git clone https://github.com/Joshua-Ashton/mini-source-sdk.git
+```
+
+Following your clone of the mini-source-sdk, navigate to the "version" you would like to build (`sdk2013-mp`, `sdk2013-sp`, `asw`). As an example, the multiplayer version will be used from now on.
+```bash
+cd mini-source-sdk/sdk2013-mp/src/
+```
+
+At this time you should now recursively git clone the VPhysics-Jolt repository into your "src" directory.
+```bash
+git clone https://github.com/Joshua-Ashton/VPhysics-Jolt.git vphysics_jolt --recursive
+```
+
+<strong>From this point onward, please follow the rest of the guide specific to your chosen build platform. The current working directory is also assumed to be `mini-source-sdk/sdk2013-mp/src/` and should be adjusted to suit your own build if need be.</strong>
+
+## Windows
+
+### CLI
+
+Continuing on windows, the next step is to the registry fix script (as administrator). This can be done by right-clicking the `fix_registry.bat` file located in `mini-source-sdk/sdk2013-mp/src/`. If your command prompt or powershell instance are already running under administrator just run the following:
+```bash
+.\fix_registry.bat
+```
+
+This final command to run before the CLI portion of this guide ends. It will generate your [Visual Studio](https://visualstudio.microsoft.com/vs/) project/solution file(s). (You may also just double-click the file `createjoltprojects.bat`)
+```bash
+.\createjoltprojects.bat
+```
+
+### Visual Studio
+
+After running `createjoltprojects.bat` you should now have a file named `jolt.sln`. Open it. After opening building may be as simple as pressing the shortcut `Ctrl+Shift+B` or right-clicking `Solution 'jolt'` bar in the Solution explorer window and pressing `Build Solution`
+
+Congratulations!
+
+The built `vphysics_jolt.dll` file should now be located in `mini-source-sdk/sdk2013-mp/game/bin/`.
+
+### Build errors
+
+Common build errors you may run into building on windows and their solutions.
+
+> #### memoverride.cpp
+
+If there is an error about involving a symbol named ` size_t _msize` or `size_t __cdecl _msize_base` try removing or commenting out definitions of <strong>both</strong>. You should leave only the code below:
+```cpp
+size_t msize( void *pMem )
+{
+	return g_pMemAlloc->GetSize(pMem);
+}
+```
+
+Where the code to be removed/commented would be:
+```cpp
+size_t __cdecl _msize_base( void *pMem ) noexcept
+{
+	return g_pMemAlloc->GetSize(pMem);
+}
+
+size_t _msize( void *pMem )
+{
+	return _msize_base(pMem);
+}
+```
+
+###### It may be wise to try different variations if the given solution does not work.
+
+## Linux
+
+Continuing on linux, there are a few prerequisites to building regardless of distrobution.
+- glibc v2.29+
+- gcc v8+
+- g++ v8+
+- gcc-multilib
+- g++-multilib
+- make
+
+###### Note: Attempts to compile with older versions of glibc are undocumented and probably will not work (you are still welcome to confirm this)
```

Comment:
meh

Reply:
I tried it already, they don't. Just being thorough again